katello/katello-change-hostname

```diff
@@ -0,0 +1,225 @@
+#!/usr/bin/ruby
+
+require "socket"
+require "optparse"
+require 'yaml'
+
+DEFAULT_PROGRAM = "foreman"
+DEFAULT_SCENARIO = "katello"
+
+def is_rhel_version?(version)
+  `cat /etc/redhat-release`[/(\d+\.)/].include?(version)
+end
+
+def timestamp
+  Time.now.strftime("%Y%m%d%H%M")
+end
+
+def get_hostname
+  if is_rhel_version?("6")
+    hostname = `hostname`
+  else
+    hostname = Socket.gethostname
+  end
+  hostname.chomp 
+end
+
+def has_disable_system_check_option?
+  katello_installer_rpm = `rpm -qa | grep katello-installer`
+  katello_installer_version = katello_installer_rpm[/(\d+\.)(\d+\.)(\d+)/]
+  Gem::Version.new(katello_installer_version) >= Gem::Version.new("3.2.0")
+end
+
+def yesno
+  begin
+    system("stty raw -echo")
+    str = STDIN.getc
+  ensure
+    system("stty -raw echo")
+  end
+  if str.chr.downcase == "y"
+    return true
+  elsif str.chr.downcase == "n"
+    return false
+  else
+    puts "Invalid Character. Try again: [y/n]"
+    yesno
+  end
+end
+
+options = {}
+options[:program] = DEFAULT_PROGRAM
+options[:scenario] = DEFAULT_SCENARIO
+options[:system_check] = false
+
+opt_parser = OptionParser.new do |opt|
+  opt.banner = "Usage: katello-change-hostname HOSTNAME [OPTIONS]"
+  opt.separator  ""
+  opt.separator  "Example:"
+  opt.separator  " katello-change-hostname foo.example.com -u admin -p changeme"
+  opt.separator  ""
+  opt.separator  "Options"
+
+  opt.on("-u","--username USERNAME","admin username REQUIRED") do |username|
+    options[:username] = username
+  end
+
+  opt.on("-p","--password PASSWORD","admin password REQUIRED") do |password|
+    options[:password] = password
+  end
+
+  opt.on("-g","--program PROGRAM","name of the program you are modifying (defaults to #{DEFAULT_PROGRAM})") do |program|
+    options[:program] = program
+  end
+
+  opt.on("-s","--scenario SCENARIO","name of the scenario you are modifying (defaults to #{DEFAULT_SCENARIO})") do |scenario|
+    options[:scenario] = scenario
+  end
+
+
+  if has_disable_system_check_option?
+    opt.on("-d","--disable-system-checks","runs the installer with --disable-system-checks") do |system_check|
+      options[:system_check] = true
+    end
+  end
+
+  opt.on("-h","--help","help") do
+    puts opt_parser
+    exit
+  end
+end
+opt_parser.parse!
+
+STDOUT.puts("This will modify your system. Have you taken the necessary precautions (backups, snapshots, etc...) and  want to proceed with changing your hostname? [y/n]")
+response = yesno
+unless response
+  STDOUT.puts "Hostname change aborted, no changes have been made to your system"
+  exit
+end
+
+raise 'Must run as root' unless Process.uid == 0
```

Comment:
This might be best at the top, exit out immediately as well as being obvious to find at the top. Sort of like a guard clause in a method.